tests/constraints.txt

```diff
@@ -7,17 +7,18 @@ chardet==4.0.0
 cloudpickle==1.6.0
 dataclasses==0.6
 decorator==5.0.7
+Deprecated==1.2.12
 dm-tree==0.1.6
 flatbuffers==1.12
 gast==0.3.3
-google-auth==1.28.1
+google-auth==1.29.0
 google-auth-oauthlib==0.4.4
 google-pasta==0.2.0
-gpflow==2.1.4
+gpflow==2.1.5
```

Comment:
I think upgrading GPflow should be left to some other PR, probably it won't break any of the code, but just in case...

Reply:
bear in mind this is the constraints for CI, not for the library itself, so this number isn't used by users

Reply:
> Looks good so far... why some notebooks didnt appear in manual docs build at the end?

A change to docutils meant that nbsphinx was automatically ignoring notebook files with lines over some (not very big) limit! Upgrading nbsphinx fixed it.

Reply:
gpflow doesn't use semver. Patch versions are meant to be backwards compatible

Reply:
We already pin gpflow to 2.1.x in setup.py, so I would personally be happy with this update. Also I'd be very sad if the tests didn't catch a break.

Reply:
ok, if patches should be backward compatible lets go with the upgrade